modules/gdscript/tests/test_completion.h

```
@@ -0,0 +1,200 @@
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_COMPLETION_H
+#define TEST_COMPLETION_H
+
+#ifdef TOOLS_ENABLED
+
+#include "core/io/config_file.h"
+#include "core/io/dir_access.h"
+#include "core/io/file_access.h"
+#include "core/object/script_language.h"
+#include "core/variant/dictionary.h"
+#include "core/variant/variant.h"
+#include "gdscript_test_runner.h"
+#include "modules/modules_enabled.gen.h" // For mono.
+#include "scene/resources/packed_scene.h"
+
+#include "../gdscript.h"
+#include "tests/test_macros.h"
+
+#include "editor/editor_settings.h"
+#include "scene/theme/theme_db.h"
+
+namespace GDScriptTests {
+
+static bool match_option(const Dictionary p_expected, const ScriptLanguage::CodeCompletionOption p_got) {
+	if (p_expected.get("display", p_got.display) != p_got.display) {
+		return false;
+	}
+	if (p_expected.get("insert_text", p_got.insert_text) != p_got.insert_text) {
+		return false;
+	}
+	if (p_expected.get("kind", p_got.kind) != Variant(p_got.kind)) {
+		return false;
+	}
+	if (p_expected.get("location", p_got.location) != Variant(p_got.location)) {
+		return false;
+	}
+	return true;
+}
+
+static void to_dict_list(Variant p_variant, List<Dictionary> &p_list) {
+	if (p_variant.is_array()) {
```

Maybe this should be an `ERR_FAIL_COND` since AFAICT this should never be called with anything other than an array and we would probably want to know if that happens.

---

Not sure what the usage guidelines for `ERR_FAIL_COND` are. A malformed .cfg file could result in this not being an array. Not sure whether a malformed test config should trigger ERR_FAIL_COND. But failing silently is bad anyway